xpti/include/xpti_trace_framework.h

```diff
@@ -0,0 +1,108 @@
+//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+#pragma once
+#include "xpti_data_types.h"
+
+#if defined(XPTI_STATIC_LIBRARY)
+// If we are building or using the proxy
+// static library we don't export any symbols
+//
+#define XPTI_EXPORT_API
+#else
+#if defined(_WIN64) || defined(_WIN32) /* Windows */
+#ifdef XPTI_API_EXPORTS
+#define XPTI_EXPORT_API __declspec(dllexport)
+#else
+#define XPTI_EXPORT_API __declspec(dllimport)
+#endif
+#else /* Generic Unix/Linux */
+#ifdef XPTI_API_EXPORTS
+#define XPTI_EXPORT_API __attribute__((visibility("default")))
+#else
+#define XPTI_EXPORT_API
+#endif
+#endif
+#endif
+
+#define XPTI_EXTRACT_USER_DEFINED_ID(val) ((uint16_t)val & 0x007f)
+#define XPTI_EXTRACT_TOOL_ID(val) (((uint16_t)val >> 8) & 0x00ff)
+
+#ifdef __cplusplus
```

Comment:
I think you can remove these `#ifdef __cplusplus` in this file too.

Reply:
Done!